src/DGSM.jl

```diff
@@ -1,10 +1,13 @@
-
 mutable struct DGSM
```

Comment:
It would make more sense to name it as `DGSM_res` or `Res_DGSM` to indicate the struct is for the returning the result, I got confused and thought this was going to be used to construct the problem